module/zfs/arc.c

```diff
@@ -9899,7 +9901,8 @@ l2arc_log_blk_commit(l2arc_dev_t *dev, zio_t *pio, l2arc_write_callback_t *cb)
 	/* try to compress the buffer */
 	list_insert_tail(&cb->l2wcb_abd_list, abd_buf);
 	psize = zio_compress_data(ZIO_COMPRESS_LZ4,
-	    abd_buf->abd, tmpbuf, sizeof (*lb));
+	    abd_buf->abd, tmpbuf, sizeof (*lb),
+	    1 << dev->l2ad_spa->spa_max_ashift);
```

Comment:
I don't think it really matters what threshold we use for the l2arc log block, in that the impact either way will be trivial to nonexistant.  But: I think it would make more sense to use dev->l2ad_vdev->vdev_ashift here, since we know exactly which device we're writing to, and IIRC spa_max_ashift doesn't take into account L2ARC devices.